types/benches/checksummed_hex_bench.rs

```diff
@@ -0,0 +1,104 @@
+use casper_types::checksummed_hex::{decode, encode, encode_iter};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+fn encode_and_print_first_5(input: &[u8]) {
+    let encoded = encode(input);
+    format!("{:5}", encoded);
+}
+
+fn encode_iter_and_print_first_5(input: &[u8]) {
+    let encoded = encode_iter(input).take(5).collect::<String>();
+    format!("{:5}", encoded);
+}
+
+fn encode_iter_collected(input: &[u8]) -> String {
+    encode_iter(input).collect()
+}
+
+fn encode_vs_encode_iter(c: &mut Criterion) {
+    let input: Vec<u8> = (0..100_000).map(|_| rand::random::<u8>()).collect();
+    let mut group = c.benchmark_group("encode_vs_encode_iter");
+
+    group.bench_function("encode", |b| b.iter(|| encode(black_box(&input))));
+    group.bench_function("encode_iter_collected", |b| {
+        b.iter(|| encode_iter_collected(black_box(&input)))
+    });
+    group.finish();
+}
+
+fn encode_vs_encode_iter_first_5_chars(c: &mut Criterion) {
```

Comment:
Question: why are we testing this use-case specifically? Do we do it often? Encode and print first 5 characters?

Reply:
We wanted a benchmark to document a case where we were potentially wasting a lot of cycles by encoding a full hex string and then truncating the string to 5 or 10 characters. We have replaced calls to `checksummed_hex::encode()` with `base16::encode_lower()` for very short strings.

We used this benchmark to make that decision, but, now that it's been made I'm not sure that we need to keep this benchmark around.